File: Framework/EntityServiceContainer.cs
using System.Collections.Generic;
using Framework.Data_Manipulation;
using Framework.Domain;

namespace Framework
{
    public interface IEntityServiceContainer<TEntity>
        where TEntity : IDomainObject
    {
        IBaseMapper<TEntity> Mapper { get; set; }
        IRepository<TEntity> Repository { get; set; }
        IDictionary<string, IBaseQueryObject<TEntity>> QueryDictionary { get; set; }
        IIdentityMap<TEntity> IdentityMap { get; set; }
    }

    public class EntityServiceContainer<TEntity> : IEntityServiceContainer<TEntity>
        where TEntity : IDomainObject
    {
        public IBaseMapper<TEntity> Mapper { get; set; }
        public IRepository<TEntity> Repository { get; set; }
        public IDictionary<string, IBaseQueryObject<TEntity>> QueryDictionary { get; set; }
        public IIdentityMap<TEntity> IdentityMap { get; set; }
    }
}

File: Android/app/src/main/java/com/EntityServiceContainer.java
package com;

import com.Interfaces.IIdentityMapConcrete;
import com.Interfaces.IRepository;
import com.DataManipulation.BaseMapperInterfaces.IBaseMapperConcrete;
import com.DataManipulation.BaseQueryObjectInterfaces.IBaseQueryObjectConcrete;
import com.Domain.Interfaces.IDomainObject;

import java.lang.reflect.Field;
import java.util.HashMap;

/**
 * Created by aiko on 2/11/17.
 */

public class EntityServiceContainer<TEntity extends IDomainObject> {
    public IBaseMapperConcrete<TEntity> Mapper;
    public IIdentityMapConcrete<TEntity> IdentityMap;
    public IRepository<TEntity> Repository;
    public HashMap<String, IBaseQueryObjectConcrete<TEntity>> QueryDictionary;
    public HashMap<String, Field> PrimitiveFields;
}

